### PR TITLE
Improve OIDC JWT Retrieval Documentation

### LINKED ISSUE #35340: Improve OIDC provider JWT retrieval documentation
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/.github/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

[Requesting the JWT using environment variables](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers#requesting-the-jwt-using-environment-variables)

### What part(s) of the article would you like to see updated?

The current documentation for manually requesting a JSON Web Token (JWT) from the GitHub OIDC provider has several issues:
1. The name of environment variable in the script is not valid ( `ACTIONS_RUNTIME_TOKEN` must be `ACTIONS_ID_TOKEN_REQUEST_TOKEN ` )
2. The whole node script step is redundant.
3. `jq` availability is checked after it is actually used.
4. Printing tokens / signature to logs has potential security risks and is not useful.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by NeoCat**
### Why:

Closes: #35340

### What's being changed (if available, include any code snippets, screenshots, or gifs):

This pull request addresses several issues in the current documentation for manually requesting a JSON Web Token (JWT) from the GitHub OIDC provider. The updates focus on improving clarity, security, and accuracy.

Changes:

- Replaced incorrect environment variable name with proper reference to `ACTIONS_ID_TOKEN_REQUEST_TOKEN`.
- Removed the direct logging of the retrieved JWT to avoid potential exposure in logs.
- Simplified and clarified the example code by removing redundant node.js step and cleaned up the `curl` lines.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Review by Obada1989 — COMMENTED** [review]
NeoCat:patch-1

**Comment by Obada1989** [comment]
دمج القرش

**Comment by Obada1989** [comment]
الالتزام بالعمل هو شيء جوهري لكن احيانا تكون الاختبارات بحاجة الى توثيق

File: content/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers.md
Comment (Obada1989) [inline content/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers.md:1]: قابلة للتحسين ساعمل على ذلك لكن شكرا لله اللذي سخرنا لخدمة المجتمع

**Review by Obada1989 — COMMENTED** [review]
content/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers.md

**Comment by nguyenalex836** [comment]
@NeoCat Thanks so much for opening a PR! I'll get this triaged for review ✨

**Comment by nguyenalex836** [comment]
@NeoCat Thank you for your patience while our SME team reviewed! Unfortunately, we are unable to accept this PR for the following reasons stated by our SME team:

> 1. The purpose of this section is to demonstrate requesting using the token using environment variables, which we provide two examples.
> 2. This PR replaces that content with a request to retrieve the token. That is unfortunately not the purpose of this section.
> 3. We are unsure what is meant regarding printing the token and signature. The signature ensures that the token has not been tampered with. When a JWT is signed, the signature is created using the header and payload of the token along with a secret key. We are unsure how displaying that value in the logs is a security threat.

Thank you again for your willingness to contribute to GitHub Docs 💛

**Pull request closed without merging**